io/io/src/TFileMerger.cxx

```diff
@@ -892,8 +955,11 @@ Bool_t TFileMerger::PartialMerge(Int_t in_type)
    } else {
       // Close or write is required so the file is complete.
       if (in_type & kIncremental) {
-         fOutputFile->Write("",TObject::kOverwrite);
+         if (!(in_type & kDelayWrite))
+            fOutputFile->Write("",TObject::kOverwrite);
       } else {
+         if (type & kIncremental)
+            fOutputFile->Write("",TObject::kOverwrite);
```

Comment:
What if `in_type & kDelayWrite`?

Reply:
if in_type is not incremental but type is incremental we are now in the case where the user "explicitly" request a non-incremental merge but we still have internally an incremental merge.  Because the user did not request the incremental merge they also probably do not to a final Write of the file and thus not doing the write here would lead to data lose ...

adding this as a comment ...